youtube_dl/extractor/senateisvp.py

```diff
@@ -5,149 +5,169 @@
+        },
     }, {
-        # From http://www.c-span.org/video/?96791-1
-        'url': 'http://www.senate.gov/isvp?type=live&comm=banking&filename=banking012715',
-        'only_matching': True,
+        'url': 'https://www.senate.gov/isvp/?comm=foreign&type=archv&stt=0&filename=foreign080322&auto_play=false&wmode=transparent&poster=https%3A%2F%2Fwww%2Eforeign%2Esenate%2Egov%2Fthemes%2Fforeign%2Fimages%2Fvideo%2Dposter%2Dflash%2Dfit%2Epng',
+        'info_dict': {
+            'id': 'foreign080322',
+            'ext': 'mp4',
+            'title': 'foreign080322',
+            'thumbnail': 'https://www.foreign.senate.gov/themes/foreign/images/video-poster-flash-fit.png',
+        },
     }]
 
     @staticmethod
+    # returns url from an iframe
     def _search_iframe_url(webpage):
         mobj = re.search(
-            r"<iframe[^>]+src=['\"](?P<url>https?://www\.senate\.gov/isvp/?\?[^'\"]+)['\"]",
+            r'''<iframe\b[^>]+\bsrc\s*=\s*(['"])(?P<url>https?://www\.senate\.gov/isvp/?\?(?:(?!\1)\S)+)''',
             webpage)
         if mobj:
             return mobj.group('url')
 
+    # returns stream_number, stream_domain, stream_id, and msl3
     def _get_info_for_comm(self, committee):
-        for entry in self._COMM_MAP:
-            if entry[0] == committee:
-                return entry[1:]
+        return self._COMM_MAP[committee][0:]
 
     def _real_extract(self, url):
+        # smuggled_data may contain a forced title that should be used
         url, smuggled_data = unsmuggle_url(url, {})
-
         qs = compat_parse_qs(re.match(self._VALID_URL, url).group('qs'))
-        if not qs.get('filename') or not qs.get('type') or not qs.get('comm'):
-            raise ExtractorError('Invalid URL', expected=True)
 
-        video_id = re.sub(r'.mp4$', '', qs['filename'][0])
+        # error handling for invalid URL - specify what
+        if not qs.get('comm'):
+            raise ExtractorError('''Invalid URL. Missing committee in query parameters.''', expected=True)
+        if not qs.get('filename'):
+            raise ExtractorError('''Invalid URL. Missing filename in query parameters.''', expected=True)
 
-        webpage = self._download_webpage(url, video_id)
+        committee = qs.get('comm')[0]
+        filename = qs.get('filename')[0]
+        video_id = re.sub(r'\.mp4$', '', filename)
 
-        if smuggled_data.get('force_title'):
-            title = smuggled_data['force_title']
-        else:
-            title = self._html_search_regex(r'<title>([^<]+)</title>', webpage, video_id)
-        poster = qs.get('poster')
-        thumbnail = poster[0] if poster else None
+        # there is no point in pulling the title from the webpage since it always defaults to 'Integrated Senate Player'
+        title = smuggled_data.get('force_title') or filename
+        stream_number, stream_domain, stream_id, msl3 = self._get_info_for_comm(committee)
```

Comment:
If getting the committee to stream info map from the page:
```suggestion
        # get the lookup table from the page
        stream_info = '[%s]' % (self._search_regex(
            r'\bnew\s+Array\s*\(([^)]+)',
            webpage, 'stream info', fatal=False) or '', )
        if stream_info:
            stream_info = self._parse_json(stream_info, video_id, transform_source=js_to_json, fatal=False)
        if stream_info:
            self._COMM_MAP = stream_info
```